### PR TITLE
drivers: can: shell: add command for configuring a CAN controller

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add shell command for configuring the mode and bitrate of a CAN controller.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

### CODE REVIEW

File: drivers/can/can_shell.c [hunk 1]
```diff
@@ -12,8 +12,9 @@
 
 static struct zcan_work work;
 
-static inline int read_options(const struct shell *shell, int pos, char **argv,
```
Comment (alexanderwachter) [inline drivers/can/can_shell.c:15]: This function was meant to only read the options for a frame actually.

**Review by alexanderwachter — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: drivers/can/can_shell.c (hunk 1 above)
Comment (henrikbrixandersen) [reply drivers/can/can_shell.c:15]: Okay. I have renamed it and implemented a separate function for reading the config options.

**Review by alexanderwachter — APPROVED** [review]

**Comment by alexanderwachter** [comment]
Thanks!

**Pull request merged**